In [3]:
import pandas as pd
from sklearn.model_selection import train_test_split

In [4]:
credit = pd.read_csv('/content/creditcard.csv')

In [5]:
count_zero = (credit['Class'] == 0).sum()
count_one = (credit['Class'] == 1).sum()
print("Liczba zer w kolumnie 'class':", count_zero)
print("Liczba jedynek w kolumnie 'class':", count_one)

Liczba zer w kolumnie 'class': 284315
Liczba jedynek w kolumnie 'class': 492


In [6]:
credit["Class"].value_counts()

0    284315
1       492
Name: Class, dtype: int64

In [7]:
X = credit.drop(columns='Class')
y = credit['Class']

In [8]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [9]:
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score, precision_score, recall_score


In [10]:
knn = KNeighborsClassifier(n_neighbors=3)
knn.fit(X_train, y_train)
y_pred = knn.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
y_pred = knn.predict(X_test)

In [11]:

accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)

In [12]:
print("Accuracy: ", accuracy)
print("Precision: ", precision)
print("Recall: ", recall)

Accuracy:  0.9985955549313578
Precision:  1.0
Recall:  0.11764705882352941


In [13]:
from sklearn.metrics import confusion_matrix

# Obliczenie macierzy pomyłek
conf_matrix = confusion_matrix(y_test, y_pred)

# Wyświetlenie macierzy pomyłek
print("Macierz pomyłek:")
print(conf_matrix)

Macierz pomyłek:
[[85307     0]
 [  120    16]]


chcemy aby te 120 było jak najmniejsze bo to niewykryte przestępstwa

In [14]:
pip install -U imbalanced-learn

In [17]:
from collections import Counter
counter = Counter(y)

In [15]:
from imblearn.over_sampling import RandomOverSampler
from imblearn.over_sampling import SMOTE

oversample = SMOTE()
X_train_os, y_train_os = oversample.fit_resample(X_train, y_train)


NameError: name 'Counter' is not defined

In [ ]:
knn = KNeighborsClassifier(n_neighbors=3)
knn.fit(X_train_os, y_train_os)
y_pred = knn.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
y_pred = knn.predict(X_test)

In [ ]:
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
print("Accuracy: ", accuracy)
print("Precision: ", precision)
print("Recall: ", recall)

In [ ]:

conf_matrix = confusion_matrix(y_test, y_pred)

# Wyświetlenie macierzy pomyłek
print("Macierz pomyłek:")
print(conf_matrix)